### PR TITLE
Fix cred type display on form initialization

### PR DESCRIPTION
**Pull request by jakemcdermott**
##### SUMMARY
For #9636 and https://github.com/ansible/awx/issues/8828

### CODE REVIEW

**Comment by nixocio** [comment]
hey @jakemcdermott, I got this code locally and I noticed the following.

1 - Editing a credential type it will trigger the form validation.

image

2 - Attempting to save a edited credential type.

image

On devel the the form validation is not triggered once the credential type is modified, and the following error is raised once one attempt to save a modified credential type.

save_validation

**Comment by jakemcdermott** [comment]
@nixocio 
> 2 - Attempting to save a edited credential type.

This should be fixed in the latest commit - can you try it again with the latest? I'll check on the other two.

**Comment by nixocio** [comment]
> @nixocio
> 
> > 2 - Attempting to save a edited credential type.
> 
> This should be fixed in the latest commit - can you try it again with the latest? I'll check on the other two.

I just tested this. It is fixed.

**Comment by unlikelyzero** [comment]
just visual tests on this one

**Comment by jakemcdermott** [comment]
cc @nixocio @unlikelyzero 

re: issues 1 and 3 mentioned [here](https://github.com/ansible/awx/pull/9662#issuecomment-804886235)

I added a commit to this PR that disables the credential-type select on the credential edit forms. Allowing the field value to be changed invites a lot of complexity and bugs. Moreover, there doesn't seem to be a good reason to let users change it in the first place given that [you can't actually modify a credential's type after it's been created.](https://github.com/ansible/awx/blob/devel/awx/api/serializers.py#L2675) Users will _always_ get an error when they try to do this

Screenshot from 2021-03-24 12-24-13

File: awx/ui_next/src/screens/Credential/shared/CredentialForm.jsx [hunk 1]
```diff
@@ -27,6 +29,7 @@ const Select = styled(PFSelect)`
   ul {
     max-width: 495px;
   }
+  ${props => (props.isDisabled ? `cursor: not-allowed` : null)}
```
Comment (nixocio) [inline awx/ui_next/src/screens/Credential/shared/CredentialForm.jsx:32]: 👍

**Review by nixocio — APPROVED** [review]
I like this new UX! and the reported issue is fixed.

**Comment by unlikelyzero** [comment]
We need a single new visual test for the hover-over disabled 'edit' component

**Comment by jakemcdermott** [comment]
regate

**Pull request merged**